nautobot/ui/src/components/Navbar.js

```diff
@@ -41,6 +45,35 @@ export function Navbar() {
     const location = useLocation();
     const { data: readyRoutes } = useGetNewUIReadyRoutesQuery();
 
+    const [searchParams, setSearchParams] = useSearchParams();
+
+    const changeHandler = (event) => {
+        let filters = [];
+        if (event.target.value !== "") {
+            filters = [["q", event.target.value]];
+        }
+        setSearchParams([
+            ...filters,
+            ...[...searchParams].filter(
+                ([searchParamLabel]) =>
+                    !FILTER_RESET_QUERY_PARAMS.includes(searchParamLabel) &&
+                    searchParamLabel !== "q"
+            ),
+        ]);
+    };
+
+    const debouncedOnChangeSearchBox = useMemo(() => {
+        return debounce(changeHandler, 100);
+    }, []); // eslint-disable-line react-hooks/exhaustive-deps -- For some reason, eslint doesn't like this dependency array, but it's needed to prevent the debounce from being recreated on every render.
```

Comment:
There is a good reason for that - every time the component renders, i.e. it's called (because it's a function), a new `changeHandler` function is created. Function creation can be avoided with memoization provided by some of the React hooks, `useCallback` in this case. Moreover, with dependency array set to `[]`, this code now has a bug, which is caused by this debounce function holding the initial `changeHandler` reference and results in `searchParams` not being updated in it. You can test it, for example, first by changing the page or page size and then searching - you should see the old (initial) params restored. To avoid it, you need to set a valid dependency array here

Reply:
@norbert-mieczkowski-codilime Can we discuss tomorrow morning?